In [47]:
import sys
import numpy as np
import matplotlib.pyplot as plt
import scipy.stats as stats
import numpy as np

## Problem size 100

In [48]:
filename = '../final_set/solutions_100.npz'
data = np.load(filename)
data = data["solutions"]
a = data[0, :]
b = data[1, :]
c = data[2, :]

solutions = np.concatenate((a, b, c))

mean = np.mean(solutions)
std_dev = np.std(solutions)

gurobi_result = 70598781.54841062

z_score = (gurobi_result - mean) / std_dev

print("Mean:", mean)
print(f"Standard Deviation: {std_dev:.2f}")
print(F"Gurobi Result: {gurobi_result:.2f}")
print(f"Z-Score: {z_score:.2f}")


Mean: 74145039.80726531
Standard Deviation: 3767986.03
Gurobi Result: 70598781.55
Z-Score: -0.94


A z-score of approximately *−0.943* means that Gurobi's result is about 0.943 standard deviations below the mean of your experimental results.


Since ∣z∣<1, Gurobi's result is within one standard deviation of the mean of your experimental results.

This suggests that Gurobi's result is well within the range of typical variability observed in your optimization experiments.

## Problem size 200

In [49]:
filename = '../final_set/solutions_200.npz'
data = np.load(filename)
data = data["solutions"]
a = data[0, :]
b = data[1, :]
c = data[2, :]

solutions = np.concatenate((a, b, c))

mean = np.mean(solutions)
std_dev = np.std(solutions)

gurobi_result = 137535570.41863775

z_score = (gurobi_result - mean) / std_dev

print("Mean:", mean)
print(f"Standard Deviation: {std_dev:.2f}")
print(F"Gurobi Result: {gurobi_result:.2f}")
print(f"Z-Score: {z_score:.2f}")


Mean: 150717433.02888843
Standard Deviation: 7328286.49
Gurobi Result: 137535570.42
Z-Score: -1.80


#### Interpretation: 

A z-score of −1.80 means that the data point is 1.80 standard deviations below the mean.

#### Implication: 
This indicates that the result is somewhat unusual but not extremely rare. In a normal distribution, about 90% of the data lies within 1.645 standard deviations of the mean, so a result with a z-score of −1.80 is less common but still within the expected range for many datasets.

#### Percentile: 
Approximately 3.59% of the data in a standard normal distribution lies below a z-score of −1.80. This means the Gurobi result is lower than about 96.41% of your experimental results.

#### Summary

Since the Gurobi result had a z-score of −1.80, it is significantly lower than the mean but not an extreme outlier. It indicates that the result is within a plausible range but on the lower end.

## Problem size 400

In [50]:
filename = '../final_set/solutions_400.npz'
data = np.load(filename)
data = data["solutions"]
a = data[0, :]
b = data[1, :]
c = data[2, :]

solutions = np.concatenate((a, b, c))

mean = np.mean(solutions)
std_dev = np.std(solutions)

gurobi_result = 270662192.8389

z_score = (gurobi_result - mean) / std_dev

print("Mean:", mean)
print(f"Standard Deviation: {std_dev:.2f}")
print(F"Gurobi Result: {gurobi_result:.2f}")
print(f"Z-Score: {z_score:.2f}")


Mean: 314004768.39694804
Standard Deviation: 12478900.56
Gurobi Result: 270662192.84
Z-Score: -3.47


#### Interpretation: 

A z-score of −3.47 means that the data point is 3.47 standard deviations below the mean.

#### Implication: 
This is a very unusual result. In a normal distribution, about 99.73% of the data lies within three standard deviations of the mean, so a result with a z-score of −3.47 is extremely rare.

#### Percentile: 
Approximately 0.02% of the data in a standard normal distribution lies below a z-score of −3.47. This means the Gurobi result is lower than about 99.98% of your experimental results.

#### Summary:
Since the Gurobi result had a z-score of −3.47, it would be considered an extreme outlier. This suggests that the Gurobi result is exceptionally lower than the typical results from our experiments, indicating a significant difference.

## Problem size 800

In [51]:
gurobi_result = 547200537.3802

## Problem size 1600

In [52]:
## Gurobi founs no answer, since it ran out of memory